lib/Sema/TypeCheckGeneric.cpp

```diff
@@ -189,7 +202,7 @@ OpaqueResultTypeRequest::evaluate(Evaluator &evaluator,
       }
 
       assert(!constraintType->hasArchetype());
-      requirements.emplace_back(kind, paramType, constraintType);
+      requirements.emplace_back(kind, paramType, constraintType);}
```

Comment:
Hmm, I don't think we want this bit of code inside an `OpaqueReturnTypeRepr` condition. We want to add a new requirement for implicit opaque return type reps too. The only difference is the `constraintType` will be resolved directly from the collected `IdentTypeRepr` or a `CompositionTypeRepr` instead of the constraint type representation of an `OpaqueReturnTypeRepr`.